### ISSUE TITLE
Cover entity is missing, error attached

### ISSUE THREAD

**Issue by jman311** [corporategoth/ha-powerpetdoor#7]
After recent HA updates, my automation that uses this integration started having problems.  Further inspection revealed that the door cover entity was no longer always reporting "open", "opening", "closed" events as they occurred back to HA. 

I then noticed the integration reported the the primary dog door "cover" entity was no longer being provided by the integration.  I tried reinstalling the integration but the cover entity is still reported as missing.

Only 7 entities are available provided by the integration now (there were 8):
- sensor.power_pet_door
- switch.power_pet_door_auto
- sensor.power_pet_door_battery
- button.power_pet_door_cycle
- switch.power_pet_door_inside_sensor
- switch.power_pet_door_outside_sensor
- switch.power_pet_door_power

The "cover.power_pet_door" entity is now missing entirely.

 - Home Assistant Version: 2023.7.3
 - Power Pet Door Integration Version: 0.3.2
 - Power Pet Door Model: Large, Wifi

I turned on debugging and restarted,  This is coming up in my system log:

---
Logger: homeassistant.components.cover
Source: custom_components/powerpetdoor/cover.py:59
Integration: Cover ([documentation](https://www.home-assistant.io/integrations/cover), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+cover%22))
First occurred: 3:56:23 PM (1 occurrences)
Last logged: 3:56:23 PM

Error while setting up powerpetdoor platform for cover
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 353, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/powerpetdoor/cover.py", line 204, in async_setup_entry
    PetDoor(hass=hass,
  File "/config/custom_components/powerpetdoor/cover.py", line 59, in __init__
    update_interval=timedelta(seconds=update_interval) if update_interval else None)
                    ^^^^^^^^^
NameError: name 'timedelta' is not defined
---

Thanks,
Josh

**Comment by corporategoth** [comment]
Try this on 0.4.0 - it should be fixed.

**Comment by jman311** [comment]
Thanks for the update!  Looks like you added a lot.

I am getting a new error on 0.4.0, an integration setup failure upon restart of HA:


Logger: homeassistant.config_entries
Source: custom_components/powerpetdoor/__init__.py:76
Integration: Power Pet Door (documentation, issues)
First occurred: 12:53:33 AM (1 occurrences)
Last logged: 12:53:33 AM

Error setting up entry Power Pet Door for powerpetdoor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 390, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/powerpetdoor/__init__.py", line 76, in async_setup_entry
    entry.options[ent["field"]] = entry.data[ent["field"]]
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^
TypeError: 'mappingproxy' object does not support item assignment

**Comment by rdgerken** [comment]
Hi - great work!  I love this integration.

I just wanted to add that I'm getting the same "mappingproxy" error reported by jman311. Please let me know if I can help debug.

**Comment by corporategoth** [comment]
0.4.1 should fix this.  Sorry 'bout that ;)

**Comment by jman311** [comment]
Good morning!  Thank you for the new version update.  I tried installing this and get a new error.  Sorry to be bringing you another bug report!

`This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/powerpetdoor/__init__.py:81
Integration: Power Pet Door (documentation, issues)
First occurred: 9:24:52 AM (1 occurrences)
Last logged: 9:24:52 AM

Error setting up entry Power Pet Door for powerpetdoor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 390, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/powerpetdoor/__init__.py", line 81, in async_setup_entry
    options[ent["field"]] = entry.get("default")
                            ^^^^^^^^^
AttributeError: 'ConfigEntry' object has no attribute 'get'
`

**Comment by corporategoth** [comment]
Ugh - it's hard to debug code I'm not hitting.

0.4.1 retracted, and 0.4.2 is born :)

**Comment by jman311** [comment]
As a SQL developer I can relate.   I get this error with 0.4.2.  Looks like same object different attribute:


This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/powerpetdoor/__init__.py:89
Integration: Power Pet Door (documentation, issues)
First occurred: 3:48:20 PM (1 occurrences)
Last logged: 3:48:20 PM

Error setting up entry Power Pet Door for powerpetdoor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 390, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/powerpetdoor/__init__.py", line 89, in async_setup_entry
    await entry.async_update_entry(entry, data=data, options=options)
          ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ConfigEntry' object has no attribute 'async_update_entry'

**Comment by corporategoth** [comment]
Rather than releasing another version, can you try updating the line (89 of init.py) and change:
```
await entry.async_update_entry(entry, data=data, options=options)
```

to:
```
await hass.config_entries.async_update_entry(entry, data=data, options=options)
```

See if you still get an error.

**Comment by jman311** [comment]
Made the update, here is the latest:

This error originated from a custom integration.

Logger: homeassistant.config_entries
Source: custom_components/powerpetdoor/__init__.py:89
Integration: Power Pet Door (documentation, issues)
First occurred: 8:07:02 AM (1 occurrences)
Last logged: 8:07:02 AM

Error setting up entry Power Pet Door for powerpetdoor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 390, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/powerpetdoor/__init__.py", line 89, in async_setup_entry
    await hass.config_entries.async_update_entry(entry, data=data, options=options)
TypeError: object bool can't be used in 'await' expression

**Comment by corporategoth** [comment]
Huh - apparently it's not an async function.

Take out the `await` in the new line, and see if it's all good.

**Comment by jman311** [comment]
Alright, I removed the "await" and things seem to be ok now.  No errors, the door is responding again and controls appear to be working.  W00t!  

Thank you so much for fixing this.

**Comment by corporategoth** [comment]
OK, released 0.4.3 for this
